index.js

```diff
@@ -14,18 +14,16 @@
  \*───────────────────────────────────────────────────────────────────────────*/
 'use strict';
 
-var async = require('async'),
-  Setup = require('./setup'),
+var Promiz = require('./lib/promise'),
+  Configure = require('./lib/configure'),
+  Setup = require('./lib/setup'),
```

Comment:
That was actually my original refactor (export main function plus `Configure` and `Setup`). But I didn't like that calling `Nemo` and calling `Nemo.Setup` returned the same thing (the nemo object and driver instance). It felt cleaner to override the main constructor. Also, re: argument processing. I may have just made it look complex with the way I wrote the argument resolution code :)